### PR TITLE
feat: implement removing favorite room tags

### PR DESCRIPTION
**Pull request by domw30**
### What does this do?
- removes room from favorites (matrix)
- handles room unfavorited updates (saga)
- adds and handles action for unfavoriting room
- updates the publishing of room tag updates

### Why are we making this change?
- to be able to mark rooms (conversations) NOT as favorites (unfavoriting) for an enhanced user experience

### How do I test this?
- run tests as usual.
- alternatively, you could call the `removeRoomFromFavorites` and check the tag event is logged and the event content tags no longer contain the `m.favorite` tag.

#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

REMOVE Favorite Demo (once wired up) with rough three dot menu

https://github.com/zer0-os/zOS/assets/39112648/98819d9c-f182-4ca1-aa96-852e45d874c9

### CODE REVIEW

File: src/lib/chat/matrix-client.ts [hunk 1]
```diff
@@ -936,7 +942,7 @@ export class MatrixClient implements IChatClient {
     if (isFavoriteTagAdded) {
       this.events.roomFavorited(roomId);
     } else {
-      console.log('room unfavorited');
+      this.events.roomUnfavorited(roomId);
```
Comment (dalefukami) [inline src/lib/chat/matrix-client.ts:945]: I'm curious about this TAG event. Is this going to get fired whenever _any_ tag gets changed? Is there a way to know which tag actually changed rather than firing off these Favorite events even if an unrelated tag changes?

File: src/store/channels/saga.ts [hunk 2]
```diff
@@ -110,6 +110,18 @@ export function* onFavoriteRoom(action) {
...
 }
 
+export function* onUnfavoriteRoom(action) {
+  const { roomId } = action.payload;
+
+  const channel = yield select(rawChannelSelector(roomId));
+
+  if (!channel) {
+    return;
+  }
```
Comment (dalefukami) [inline src/store/channels/saga.ts:120]: Unnecessary check?

File: src/store/channels/saga.ts [hunk 3]
```diff
@@ -110,6 +110,18 @@ export function* onFavoriteRoom(action) {
...
+
+  yield call(removeRoomFromFavorites, roomId);
```
Comment (dalefukami) [inline src/store/channels/saga.ts:122]: We should probably wrap these in try/catch statements. What if Matrix fails to execute this...what should we tell the user?

**Review by dalefukami — APPROVED** [review]

File: src/store/channels/saga.ts (hunk 2 above)
Comment (domw30) [reply src/store/channels/saga.ts:120]: Removed the unnecessary checks

File: src/store/channels/saga.ts (hunk 3 above)
Comment (domw30) [reply src/store/channels/saga.ts:122]: Follow up was going to include some form of basic error state for either of these saga functions failing. For this PR I have now added try/catch statements, simply logging error. Shall implement something for the user as a follow up

File: src/lib/chat/matrix-client.ts (hunk 1 above)
Comment (domw30) [reply src/lib/chat/matrix-client.ts:941]: Yeah as discussed previously, our current interaction with tags is limited too managing favorites, correct in pointing out this firing whenever there is a tag change - the tag structure doesn't inherently differentiate between the addition or removal of specific tags—it simply presents the current state. This approach would indeed require reevaluation and fine-tuning should our application's tag-related features expand beyond favorites
